src/App.js

```diff
@@ -1,19 +1,49 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
+
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages)
+  const [totalLikes, setTotalLikes] = useState(0)
+
+  
+
+  const updateChatData = (liked, id) => {
+    const entries = chatData.map((entry) => {
+      if (entry.id ===id) {
+        return {...entry, liked}
+      }
+      return entry;
+    })
+    const newTotalLikes = entries.reduce((acc, obj) => {
+      if (obj.liked) {
+        acc += 1;
+      }
+      return acc;
+    }, 0)
+    setChatData(entries);
+    setTotalLikes(newTotalLikes);
```

Comment:
Since you are using state for the heart tallies, this works well. If you didn't want to use state however, you could absolutely separate these two functions so that you have a totalLikesCounter function that can be run when the app renders. That would remove state making things a little simpler!